file_walk.go

```diff
@@ -130,73 +133,88 @@ func (v *Visitor) Visit(node ast.Node) ast.Visitor {
 	return v
 }
 
-// coverage returns the number of covered and total lines.
-func (f *Function) coverage(profile *cover.Profile) (int, int) {
+type funcInfo struct {
+	fileName       string
+	pkgFileName    string
+	functionName   string
+	startLine      int
+	endLine        int
+	uncoveredLines int
+}
+
+func getFunctionInfo(profiles []*cover.Profile) ([]*funcInfo, int, int, error) {
 	total := 0
 	covered := 0
+	var funcInfos []*funcInfo
 
-	for _, b := range profile.Blocks {
-		if b.StartLine > f.endLine || (b.StartLine == f.endLine && b.StartCol >= f.endCol) {
-			break
-		}
-
-		if b.EndLine < f.startLine || (b.EndLine == f.startLine && b.EndCol <= f.startCol) {
-			continue
+	for _, profile := range profiles {
+		filenames, err := getFileNames()
+		if err != nil {
+			return nil, 0, 0, err
 		}
+		for _, filename := range filenames {
+			functions, err := getFunctions(filename)
+			if err != nil {
+				return nil, 0, 0, err
+			}
+			for _, f := range functions {
+				c, t := f.coverage(profile)
+
+				fi := &funcInfo{
+					fileName:       filename,
+					pkgFileName:    filename,
+					functionName:   f.name,
+					startLine:      f.startLine,
+					endLine:        f.endLine,
+					uncoveredLines: t - c,
+				}
+
+				funcInfos = append(funcInfos, fi)
+				total += t
+				covered += c
+			}
 
-		total += b.NumStmt
-
-		if b.Count > 0 {
-			covered += b.NumStmt
 		}
 	}
 
-	if total == 0 {
-		total = 1
-	}
-
-	return covered, total
+	return funcInfos, covered, total, nil
 }
 
-func GetCoverageData(filePath string) ([]*funcInfo, int, int, error) {
-	profiles, err := getProfiles(filePath)
-	if err != nil {
-		return nil, 0, 0, err
-	}
-
-	fi, covered, total, err := getFunctionInfo(profiles)
+// walkDir gets list of go files in repo
+func walkDir() ([]string, error) {
+	var filesDir []string
+	// get mod dir
+	modeFile, err := getMod()
```

Comment:
Could you please explain this? Is it related to `go.mod`?

Reply:
using `go env` and `GOMOD` the root directory for the current repo is being set and using the base path from the root of the directory using file walk func files in each dir is collected and used further, this will have the correct path as the root directory will be basepath of the repo.